### PR TITLE
filesystem-spec 2022.10.0

### PR DESCRIPTION
**Pull request by SC426**
s3fs 2020.10 --> filesystem-spec 2022.10.0

Jira ticket: https://anaconda.atlassian.net/browse/PKG-154
Upstream repo: https://github.com/fsspec/filesystem_spec
Changelog: https://github.com/fsspec/filesystem_spec/blob/2022.10.0/docs/source/changelog.rst

### CODE REVIEW

**Review by skupr-anaconda — APPROVED** [review]

**Review by varlackc — COMMENTED** [review]
There seem to be a large set of dependencies mentioned in the upstream.
The question might be how do we differenciate between the optionan and the mandatory dependencies. 
https://github.com/fsspec/filesystem_spec/blob/2021.10.0/tox.ini#L9&L36

The recipe in conda-forge seem to ignore as well a lot of the dependencies mentioned but it is not so clear why. 
https://github.com/conda-forge/filesystem-spec-feedstock/blob/main/recipe/meta.yaml

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
-{% set version = "2022.7.1" %}
+{% set version = "2022.10.0" %}
 {% set name = "fsspec" %}
-{% set sha256 = "7f9fb19d811b027b97c4636c6073eb53bc4cbee2d3c4b33fa88b9f26906fd7d7" %}
+{% set sha256 = "cb6092474e90487a51de768170f3afa50ca8982c26150a59072b16433879ff1d" %}
 
 
 package:
```
Comment (psteyer) [inline recipe/meta.yaml:6]: Try building without `jinja2`

File: recipe/meta.yaml [hunk 2]
```diff
@@ -38,10 +38,15 @@ test:
     - pip check || 1      # [win]
```
Comment (psteyer) [inline recipe/meta.yaml:37]: I would remove line 37 and 38 and replace it with `pip check`.

**Review by psteyer — APPROVED** [review]
LGTM

**Pull request merged**